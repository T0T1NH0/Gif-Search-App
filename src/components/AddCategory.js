import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [InputValue, setInputValue] = useState(String);

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
    console.log('handleInputChange activated');
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    console.log('submit done!');

    if (InputValue.trim().length > 2) {
      setCategories((c) => [InputValue, ...c]);
      setInputValue('');
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <p>{InputValue}</p>
      <input
        type='text'
        value={InputValue}
        onChange={handleInputChange}
        placeholder='Search Here!'
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
