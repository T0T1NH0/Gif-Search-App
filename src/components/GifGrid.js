import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
  const { data, loading } = useFetchGifs(category);
  return (
    <>
      <h3>{category}</h3>
      {loading && <p className='loading animate__heartBeat'>Loading...</p>}
      <div className='card-grid'>
        {data.map((img) => {
          return <GifGridItem key={img.id} {...img} />;
        })}
      </div>
    </>
  );
};
