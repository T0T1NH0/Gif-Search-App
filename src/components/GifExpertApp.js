import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

const GifExpertApp = () => {
  const [categories, setCategories] = useState(['Ousama Ranking']);

  return (
    <div>
      <h2>Gif Search</h2>
      <hr />
      <AddCategory setCategories={setCategories} />
      <ol>
        {categories.map((category) => {
          return <GifGrid category={category} key={category} />;
        })}
      </ol>
    </div>
  );
};

export default GifExpertApp;
