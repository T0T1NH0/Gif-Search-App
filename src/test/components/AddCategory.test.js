import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('pruebas en <AddCategory/>', () => {
  const setCategories = jest.fn();
  const wrapper = shallow(<AddCategory setCategories={setCategories} />);

  test('debe de mostrarse correctamente <AddCategory/>', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de cambiar la caja de texto', () => {
    const input = wrapper.find('input');
    const value = 'hello world!';

    input.simulate('change', { target: value });
    expect(wrapper.find('p').text().trim()).toBe('');
  });

  test('No debe de postear la informacion submit', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} });
  });
});
