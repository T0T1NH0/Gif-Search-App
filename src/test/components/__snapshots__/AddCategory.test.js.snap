// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <AddCategory/> debe de mostrarse correctamente <AddCategory/> 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Search Here!"
    type="text"
    value=""
  />
</form>
`;
